src/generate.rs

```diff
@@ -767,7 +772,19 @@ pub fn fields(
                     let description =
                         format!("Possible values of the field `{}`", f.name);
 
-                    if let Some(ref register) = base.register {
+                    if let (Some(ref peripheral), Some(ref register)) = (base.peripheral, base.register) {
+                        let mod1_ = peripheral.to_sanitized_snake_case();
+                        let mod2_ = register.to_sanitized_snake_case();
+                        let mod1_ = Ident::new(&*mod1_);
+                        let mod2_ = Ident::new(&*mod2_);
+
+                        mod_items.push(
+                            quote! {
+                                #[doc = #description]
+                                pub type #pc_r = super::super::#mod1_::#mod2_::#base_pc_r;
```

Comment:
nit: I think the super::super is not necessary as all the peripheral modules are in the root of the crate.